### ISSUE TITLE
column-calcs - Update bottomCalc after rowSelectionChanged

### ISSUE THREAD

**Issue by exetico** [olifolkerd/tabulator#2545]
Hi,

**So, what's up?**
First, i've mentioned everything [on Stackoverflow](https://stackoverflow.com/questions/59393020/update-bottomcalc-value-based-on-selection-if-any), however, i realised that it's actually a full-feature request.

**Request**
I would like to see an option, where column-calc (bottomCalc/topCalc) are updated, based on the column selection.

This would allow users to see a sum, by just selecting a few rows - or something simular. It also makes great sense, for users using the "copy"-function, so the bottomCalc/topCalc, where `clipboardCopySelector: "selected"` are in use.

Currently:
![image](https://user-images.githubusercontent.com/3549445/71178913-6a65eb80-226f-11ea-908c-ae9515b68db5.png)

Example of result, with this feature implemented, based on the selection:
![image](https://user-images.githubusercontent.com/3549445/71179066-b9ac1c00-226f-11ea-95b1-471a2fb76d3e.png)
_It could also be formatted in italic or so, to reflect that the current sum, is based on the selection._

And oh, btw - thanks for a brilliant solution! I'm already _almost_ in love with it :D

If you need anything else, please let me know. I think multiple people which use the copy-function, in conbination with `selectable`, would like this feature.

**Comment by olifolkerd** [comment]
Hey @exetico 

Thanks for your kind words, it is always great to hear people are enjoying using Tabulator!

That is an interesting idea you have there, i think to make it more generic i would look at adding a ***columnRecalc*** function to the table, that could then be called by anyone at anypoint allowing it to be used in more places.

I will add it to the roadmap for a future release.

Cheers

Oli :)

**Comment by exetico** [comment]
Hi Oli,

Np! Thanks for the fast reply, too.

That's nice to hear! I hope it will be implemented soon, but let's see 😁🤪

**Comment by olifolkerd** [comment]
Hey @exetico 

I have pushed the code for this feature to the 4.6 branch which will be released in a few weeks.

You will be able to trigger a recalculation of all column calculations using the new ***recalc*** function

```js
table.recalc(); //trigger recalculation of all column calcs

Cheers

Oli :)

**Comment by exetico** [comment]
I'm missing the dancing-emoji.

Thanks Oli, and Happy New Year!

![Dancing Man](https://i.pinimg.com/236x/ff/bd/27/ffbd276e46ddbed34265127e091ad128.jpg)

**Comment by exetico** [comment]
Hi @olifolkerd 

I've tried to find a release-plan, or so - so i would just like to here, if we're close to the 4.6 release? I'm sorry for spending your time on non-dev things, but i'll just like to here, if it will be in this month, or so.

And oh; **Again thanks for a great solution!** I've used Tabulator for a few other projects now. And.. It's still brilliant :-)

From time to time, i found it very difficult to find a specific feature by search-words, but, most of time, Google is helping me out. I'm not quite sure about, what could improve this. Maybe more (if there is any) keywords, to each feature :-)
